src/internal/scheduler/animationFrame.ts

```diff
@@ -50,18 +56,25 @@ export const animationFrame: SchedulerLike = {
          }, delay)
        );
     } else {
-      _queue.push(work, state, subscription);
-      if (_animationFrameId === -1) {
-        _animationFrameId = requestAnimationFrame(flushQueue);
-      }
+      // Start the animation frame loop if it's not already going
+      startAnimationFrameLoop();
+
+      // Push a task to execute onto the queue.
+      const task = { work, state, subscription };
+      _nextQueue.push(task);
+
+      // Set up the teardown
       subscription.add(() => {
-        const index = _queue.indexOf(subscription);
+        // Remove the task from the queue if it's still in there
+        // It won't be in there if it's been executed.
+        const index = _currentQueue.indexOf(task);
         if (index >= 0) {
-          _queue.splice(index - 2, 3);
+          _currentQueue.splice(index, 1);
```

Comment:
@benlesh could we eliminate these `indexOf()` and `splice()` calls if we check `task.subscription.closed` inside `flush()`?